### PR TITLE
CASSANDRA-19637 (trunk):  Makes conditions behavior on frozen and non-frozen columns consistent for null column values 

### PR DESCRIPTION
**Pull request by blerer**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java [hunk 1]
```diff
@@ -488,10 +491,15 @@ public boolean appliesTo(Row row)
             return false;
         }
 
+        private boolean isEmpty(CollectionType<?> type, Term.Terminal value)
```
Comment (ekaterinadimitrova2) [inline src/java/org/apache/cassandra/cql3/conditions/ColumnCondition.java]:
```suggestion
        private boolean isEmpty(Term.Terminal value)
```

File: test/unit/org/apache/cassandra/cql3/conditions/ColumnConditionTest.java [hunk 2]
```diff
@@ -209,66 +220,66 @@ private static void assertThrowsIRE(CheckedFunction runnable, String errorMessag
...
-        assertTrue(conditionApplies((ByteBuffer) null, EQ, (ByteBuffer) null));
+        assertTrue(appliesSimpleCondition(ONE, EQ, ONE));
+        assertFalse(appliesSimpleCondition(TWO, EQ, ONE));
+        assertFalse(appliesSimpleCondition(ONE, EQ, TWO));
+        assertFalse(appliesSimpleCondition(ONE, EQ, EMPTY_BYTE_BUFFER));
+        assertFalse(appliesSimpleCondition(EMPTY_BYTE_BUFFER, EQ, ONE));
+        assertTrue(appliesSimpleCondition(EMPTY_BYTE_BUFFER, EQ, EMPTY_BYTE_BUFFER));
+        assertFalse(appliesSimpleCondition(ONE, EQ, null));
+        assertFalse(appliesSimpleCondition(null, EQ, ONE));
+        assertTrue(appliesSimpleCondition((ByteBuffer) null, EQ, (ByteBuffer) null));
```
Comment (ekaterinadimitrova2) [inline test/unit/org/apache/cassandra/cql3/conditions/ColumnConditionTest.java]: nit:
```suggestion
        assertTrue(appliesSimpleCondition(null, EQ, null));
```

File: test/unit/org/apache/cassandra/cql3/conditions/ColumnConditionTest.java [hunk 3]
```diff
@@ -209,66 +220,66 @@ private static void assertThrowsIRE(CheckedFunction runnable, String errorMessag
...
-        assertFalse(conditionApplies((ByteBuffer) null, NEQ, (ByteBuffer) null));
+        assertFalse(appliesSimpleCondition(ONE, NEQ, ONE));
+        assertTrue(appliesSimpleCondition(TWO, NEQ, ONE));
+        assertTrue(appliesSimpleCondition(ONE, NEQ, TWO));
+        assertTrue(appliesSimpleCondition(ONE, NEQ, EMPTY_BYTE_BUFFER));
+        assertTrue(appliesSimpleCondition(EMPTY_BYTE_BUFFER, NEQ, ONE));
+        assertFalse(appliesSimpleCondition(EMPTY_BYTE_BUFFER, NEQ, EMPTY_BYTE_BUFFER));
+        assertTrue(appliesSimpleCondition(ONE, NEQ, null));
+        assertTrue(appliesSimpleCondition(null, NEQ, ONE));
+        assertFalse(appliesSimpleCondition((ByteBuffer) null, NEQ, (ByteBuffer) null));
```
Comment (ekaterinadimitrova2) [inline test/unit/org/apache/cassandra/cql3/conditions/ColumnConditionTest.java]:
```suggestion
        assertFalse(appliesSimpleCondition(null, NEQ, null));
```

**Review by ekaterinadimitrova2 — APPROVED** [review]
The comments from 4.0 apply to all branches. Left some nits for trunk here.
All this can be addressed on commit after clean CI. Every branch needs CI with the base JDK version. 4.0, 4.1 - JDK8, 5.0 and trunk - JDK11
LGTM

**Comment by blerer** [comment]
Committed manually

**Pull request closed without merging**